Проектная работа №12

maximumsweet opened the pull request:
В проекте есть: **package.json**, **.editorconfig**, **.eslintrc**, **.gitignore**.
Команда `npm run start` запускает сервер на _localhost:3000_.
Команда `npm run dev` запускает сервер на _localhost:3000_ с хот релоудом.
Адрес _localhost:3000_ загружает фронтенд проекта Mesto.
В ответ на запрос `GET localhost:3000/users` сервер вернёт JSON-объект из файла **users.json**.
В ответ на запрос `GET localhost:3000/cards` сервер вернёт JSON-объект из файла **cards.json**.
В ответ на запрос `GET localhost:3000/users/8340d0ec33270a25f2413b69`, сервер вернёт JSON-объект пользователя с переданным после **/users** идентификатором.
Если пользователя с запрошенным идентификатором нет или при запросе несуществующего адреса, API возвратит соответствующую ошибку.

maximumsweet committed:
first version

first version of project 12

talantbekov123 commented:
Здравствуйте. Благодарю за проделанную работу! К сожалению текущая реализация содержит недостаток, не позволяющий мне засчитать работу на данной итерации:
**Отлично:** Адрес localhost:3000 загружает фронтенд проекта Mesto, проект содержит все вспомогательные файлы .editorconfi .eslintrc.json .gitignore. Используете es6, стрелочные функции.

**Нужно исправить**
- Исправить ошибки линтинга (их можно увидеть запустив команду npx eslint ./)
- Если отправить GET запрос на получения пользователя которого нет в базе http://localhost:3000/users/wrongid в ответе приходит статус 200. API должен возвращать 404 статус ответа. Пример https://ibb.co/MVXZw0H

**Можно лучше**
- Обработка ошибок. Если удалить или переименовать файлы с данными (users.json и cards.json), то сервер не запускается возвразает текст `ENOENT: no such file or directory, open 'data/cards.json'`. Лучше вернуть ответ в формате json.
` res.status(500).json({ message: 'Запрашиваемый файл не найден' });`
- Добавить файл .eslintignore или  При запуске линтер все еще ругается на папку /public. Подробнее https://eslint.org/docs/user-guide/configuring#eslintignore
- С комитами все отлично, но можно немного лучше. Часто возникает соблазн написать комментарий к коммиту вида «fixes». Но помните, что лог изменений — это история ваших действий над проектом, лучше делать комиты чаще что бы вернуться обратно если вдруг что либо не работает. 
Делать коммиты, когда:
1) выполнил единицу работы
2) сделал изменения, которые потом может потребоваться отменить
Интересная статья по [ссылке](https://habr.com/ru/company/Voximplant/blog/276695/)

maximumsweet committed:
Исправлены ошибки листинга

maximumsweet committed:
Исправлены ошибки статусов ответа

1. При получении пользователя, которого нет в базе, теперь возвращается 404 статус ответа;
2. При удалении или переименовании *.json файлов с данными возвращается нормальный ответ.

maximumsweet commented:
Добрый день! Спасибо за оперативную проверку работы, и за объяснение моих ошибок. Все недочёты учтены и исправлены. Жду дальнейшей проверки.

talantbekov123 commented:
Отличная работа! Все исправлено, похвально что учли комментарии можно лучше!

**Можно лучше**
Что бы избежать повторения комментариев игнорирования нижней строки (например `no-underscore-dangle`) можно все прописать в конфиг файле `.eslint` пример ниже
```
{
  "extends": "airbnb-base",
  "rules": {"no-underscore-dangle": [2, { "allow": ["_id","__dirname"] }]}
}
```

Вторая строка позволяет игнорировать нижний слэш для "_id","__dirname". Подробнее в документации https://eslint.org/

talantbekov123 commented on routes/users.js:
```diff
@@ -0,0 +1,37 @@
+
+  fsPromises.readFile(usersPath, { encoding: 'utf8' })
+    .then((data) => {
+      // eslint-disable-next-line no-underscore-dangle
```
Отличная работа! Все исправлено, похвально что учли комментарии можно лучше!

**Можно лучше**
Что бы избежать повторения комментариев игнорирования нижней строки (например `no-underscore-dangle`) можно все прописать в конфиг файле `.eslint` пример ниже
```
{
  "extends": "airbnb-base",
  "rules": {"no-underscore-dangle": [2, { "allow": ["_id","__dirname"] }]}
}
```

Вторая строка позволяет игнорировать нижний слэш для "_id","__dirname". Подробнее в документации https://eslint.org/

maximumsweet replied:
Понял. Сейчас исправлю и отправлю ветку в master. Спасибо!

maximumsweet committed:
Добавлено новое правило в линтер

Позволяющее использовать нижнее подчёркивание коде
